src/prepare.rs

```diff
@@ -24,13 +24,30 @@ use BidiClass::*;
 /// Represented as a range of byte indices.
 pub type LevelRun = Range<usize>;
 
+
 /// Output of `isolating_run_sequences` (steps X9-X10)
+#[derive(Debug, Eq, PartialEq)]
 pub struct IsolatingRunSequence {
     pub runs: Vec<LevelRun>,
     pub sos: BidiClass, // Start-of-sequence type.
     pub eos: BidiClass, // End-of-sequence type.
 }
 
+// Enable sorting sequences by the index of the first run
+impl Ord for IsolatingRunSequence {
```

Comment:
This seems like a weird impl to expose to users of the crate, and unfortunately it's not possible to make it private.  If this is intended only for tests, maybe they should use `sort_by` instead.